components/renderers/controls/array/array-renderer.ts

```diff
@@ -122,7 +123,7 @@ class ArrayRenderer implements JSONForms.IRenderer {
                 containerDescription['submitControls'] = this.createControlsForSubmit(items, schemaPath, submitElement, services);
                 if (data == undefined) {
                     containerDescription["instance"][containerDescription['path']] = [];
-                    data = [];
+                    data = containerDescription["instance"][containerDescription['path']];
```

Comment:
Yeah, but it doesn't work. `data` need to point to the actual reference and not to a anonymous array.